src/main/java/jnr/unixsocket/DgramUnixSocketChannel.java

```diff
@@ -0,0 +1,281 @@
+package jnr.unixsocket;
+
+import jnr.constants.platform.Errno;
+import jnr.constants.platform.ProtocolFamily;
+import jnr.constants.platform.Sock;
+import jnr.constants.platform.SocketLevel;
+import jnr.constants.platform.SocketOption;
+import jnr.enxio.channels.NativeSocketChannel;
+import jnr.ffi.LastError;
+import jnr.ffi.byref.IntByReference;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.channels.Channels;
+import java.nio.channels.SelectionKey;
+
+/**
+ *
+ * DGRAM Unix Socket Channel.
+ */
+public class DgramUnixSocketChannel extends NativeSocketChannel {
+
+  enum State {
+
+    UNINITIALIZED,
+    CONNECTED,
+    IDLE,
+    CONNECTING,
+  }
+  private volatile State state;
+  private UnixSocketAddress remoteAddress = null;
+  private UnixSocketAddress localAddress = null;
+  private final Object stateLock = new Object();
+
+  /**
+   * Open a new DgramUnixSocketChannel.
+   *
+   * @return this
+   * @throws IOException if socket fails to connect
+   */
+  public static final DgramUnixSocketChannel open() throws IOException {
+    return new DgramUnixSocketChannel();
+  }
+
+  /**
+   * connect to an unixSocketAddress.
+   *
+   * @param remote unixSocketAddress
+   * @return this
+   * @throws IOException if socket fails to connect
+   */
+  public static final DgramUnixSocketChannel open(UnixSocketAddress remote) throws IOException {
+    DgramUnixSocketChannel channel = new DgramUnixSocketChannel();
+    channel.connect(remote);
+    return channel;
+  }
+
+  public static final OutputStream openOutputStream(UnixSocketAddress remote) throws IOException {
+    DgramUnixSocketChannel channel = new DgramUnixSocketChannel();
+    channel.connect(remote);
+    return Channels.newOutputStream(channel);
+  }
+
+  /**
+   * invoke native socketpair.
+   *
+   * @return an array of two DgramUnixSocketChannels
+   * @throws IOException if socket fails to connect
+   */
+  public static final DgramUnixSocketChannel[] pair() throws IOException {
+    int[] sockets = {-1, -1};
+    Native.socketpair(ProtocolFamily.PF_UNIX, Sock.SOCK_DGRAM, 0, sockets);
```

Comment:
this is basically the only difference between this and ```UnixSocketChannel```, I can add an abstract SocketChannel to capture the shared logic.